StyleCop.Analyzers/StyleCop.Analyzers.Test/DocumentationRules/SA1648UnitTests.cs

```diff
@@ -18,6 +18,40 @@ namespace StyleCop.Analyzers.Test.DocumentationRules
     /// </summary>
     public class SA1648UnitTests
     {
+        [Fact]
```

Comment:
Focus here and let the tests drive your changes in the analyzer. I assume that we need many more tests to cover all situations when it should and should not fire.

I would especially have added some tests where the base is class is not in the compiled assembly. The easiest route might be to always inherit from an external base classes (or none at all).

How about structs and records?

Reply:
> The extension Fine Code Coverage is free and easy to set up and use if you currently don't have any easy way of checking coverage. Aim for 100% coverage for your new code.

You can also send a PR, and click the link to the Codecov Report. By default, it will filter the view down to just the files which changed in the PR so you can see the coverage of code you added/changed.

Reply:
> Do you know about an analyzer where this is done (for me to better understand it)? Or could you tell a me good candidate for such base class?
> 
> I mean I think I understand what you suggest but I'm not sure if there is a convention how this is being tackled in this library. I would like to avoid reinventing the wheel if possible.

Sorry, I forgot this. The "core" assemblies are available to you automaatically in the test code, so you can for example use ArgumentException, which has multiple constructors with and without parameters. Or List or any other core type that suits your needs.